### PR TITLE
Add support for meshopt, remove html scripts, remove use_local_engine

### PR DESCRIPTION
**Pull request by slimbuck**
This PR includes the following changes:
- remove support for use_local_engine url parameter
- move script code from the index.html to script
- changes to support latest engine GlbParser callbacks
- added support for meshoptimizer (further engine work is needed for this to work fully)

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <link rel="icon" type="image/png" href="./playcanvas-logo.png" />
     <link rel="stylesheet" href="https://code.playcanvas.com/pcui.css">
     <link rel="stylesheet" href="style.css">
+    <script src="./playcanvas.js"></script>
```
Comment (willeastcott) [inline index.html:10]: Maybe just make this `https://code.playcanvas.com/playcanvas-stable.js` until @ellt92 switches the viewer to grab the engine from NPM?

File: index.html [hunk 2]
```diff
@@ -26,28 +27,6 @@
     <script src="https://code.playcanvas.com/pcui-runtime-main.js"></script>
     <script src="./src/viewer.js"></script>
     <script src="./src/controls.js"></script>
-    <script>
```
Comment (yak32) [inline index.html:29]: exactly what I needed to eliminate unsafe-inline in CSP headers

File: index.html (hunk 1 above)
Comment (slimbuck) [reply index.html:10]: Could do, but viewer may not work until next engine release (due to meshopt changes).

**Comment by yak32** [comment]
btw, if you eliminated inline script section in index.html, could you also eliminate` onload="main()"` so I can remove unsafe-inline from CSP header?

this is replacement, should be in JS file somewhere:
```
document.body.onload = function(){
    main();
};
```

**Comment by slimbuck** [comment]
@yak32 - done.

**Pull request merged**